src/audio/data_blob.c

```diff
@@ -164,13 +164,20 @@ int comp_init_data_blob(struct comp_data_blob_handler *blob_handler,
 
 int comp_data_blob_set(struct comp_data_blob_handler *blob_handler,
 		       enum module_cfg_fragment_position pos, uint32_t data_offset_size,
-		       const uint8_t *fragment, size_t fragment_size)
+		       const uint8_t *cdata_in_fragment, size_t fragment_size)
```

Comment:
This module adapter and blob handler patch seems to cause an issue with IPC4, I got this in UPX from DMIC arecord:

```
[   92.621591] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx      : 0x4400000a|0x30000058: MOD_LARGE_CONFIG_SET [data size: 88]
[   92.621903] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx reply: 0x64000009|0x30000058: MOD_LARGE_CONFIG_SET
[   92.621905] sof-audio-pci-intel-tgl 0000:00:1f.3: FW reported error: 9 - Specified resource not found
[   92.621916] sof-audio-pci-intel-tgl 0000:00:1f.3: ipc error for msg 0x4400000a|0x30000058
[   92.621920] sof-audio-pci-intel-tgl 0000:00:1f.3: sof_ipc4_set_get_data: large config set failed at offset 0: -22
[   92.621923] sof-audio-pci-intel-tgl 0000:00:1f.3: Failed to set bytes update for DMIC0 capture Iir Eq
[   92.621925] sof-audio-pci-intel-tgl 0000:00:1f.3: kcontrol 0 set up failed for widget eqiir.12.1
[   92.621930] sof-audio-pci-intel-tgl 0000:00:1f.3: widget eqiir.12.1 freed

```

Reply:
This is now fixed with change to first commit to apply the fragment treat as cdata only for IPC3 commands.